autoload/copilot_chat/api.vim

```diff
@@ -36,15 +36,43 @@ function! copilot_chat#api#async_request(message) abort
         \ l:data,
         \ l:url]
 
-  let job = job_start(l:curl_cmd, {'out_cb': function('copilot_chat#api#handle_job_output'), 'exit_cb': function('copilot_chat#api#handle_job_close'), 'err_cb': function('copilot_chat#api#handle_job_error')})
+  if has('nvim')
+    let job = jobstart(l:curl_cmd, {
+      \ 'on_stdout': function('copilot_chat#api#handle_job_output'),
+      \ 'on_exit': function('copilot_chat#api#handle_job_close'),
+      \ 'on_stderr': function('copilot_chat#api#handle_job_error'),
+      \ 'stdout_buffered': v:true,
+      \ 'stderr_buffered': v:true
+      \ })
+  else
+    let job = job_start(l:curl_cmd, {
+      \ 'out_cb': function('copilot_chat#api#handle_job_output'),
+      \ 'exit_cb': function('copilot_chat#api#handle_job_close'),
+      \ 'err_cb': function('copilot_chat#api#handle_job_error')
+      \ })
+  endif
+
   return job
 endfunction
 
-function! copilot_chat#api#handle_job_output(channel, msg) abort
-  call add(s:curl_output, a:msg)
+function! copilot_chat#api#handle_job_output(...) abort
+  if a:0 < 2
```

Comment:
Just wondering, what is the point of this check?  Is it a "just in case" thing or can it actually happen?  Feels like there should be some messaging if it's something that can actually fail.

Reply:
its more of just in case, so that we don't have an unexpected crash. In `nvim` the `jobstart` has data as second argument, so technically should never be the case.

Reply:
"Just in case" code can be very confusing for future readers as it makes it seem like it's something that can happen.  These are callbacks and so long as nvim documents that they are always called with two args, I'd remove it.

Reply:
Im with @sodapopcan. It definitely confused me a ton when I read this diff. I'd prefer to use the named variables if that isnt a hard requirement of nvim to be ellipsis args